examples/modules/example_module.cpp

```diff
@@ -105,10 +102,25 @@ int main(int argc, char** argv) {
         });
 
     Registry::register_resource(rr);
-    my_mod->add_model_from_registry(generic, m);
 
-    my_mod->start();
-    Server::start();
-    Server::wait();
+    // The `ModuleService_` must outlive the Server, so the declaration order
+    // here matters.
+    auto my_mod = std::make_shared<ModuleService_>(argv[1]);
+    Server server;
+
+    my_mod->add_model_from_registry(&server, generic, m);
+    my_mod->start(&server);
+
+    std::thread signal_wait_thread([&server, &sigset]() {
+        int sig = 0;
+        auto result = sigwait(&sigset, &sig);
+        server.shutdown();
+    });
+
+    server.start();
```

Comment:
I think there are some windows for races here, like if the signal arrives after calling `sigwait` but before `server.start` is that OK? I think it is worth considering whether the thread is started at the right place and thinking through the potential ordering issues to make sure everything is well behaved.

Reply:
> if the signal arrives after calling sigwait but before server.start is that OK?

Good point; I don't think so. `shutdown` is a no-op when the server hasn't been started yet, but the thread is only able to catch a single `SIGTERM` or `SIGWAIT`. Looking into a better ordering here (putting `server.start()` before thread creation causes an `abort trap` error).

Reply:
Decided to call `server.start()` at the top of the created thread instead of after thread creation. There are still windows where `SIGTERM` could be sent but not caught (i.e. before `sigwait`), but I think we get rid of the case described above.